src/reducers/walletMigrationArchanovaReducer.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import produce from 'immer';
```

Comment:
Not sure about this, extra library just for this piece is extra technical debt to maintain, can't we just spreads like in other reducers? 👀

Reply:
In my experience from past projects, this allows more readable code than spreading deeply nested structures. This dep is small and has a clear purpose which it already achieves, so there is little risk of building tech debt because of it.

Reply:
As an argument "for" using Immer, it is included in the [Redux Toolkit](https://redux-toolkit.js.org/usage/immer-reducers), which is officially toolkit from Redux team.

Reply:
`there is little risk of building tech debt because of it` – last famous words 😂 

but putting jokes aside lets go with it then even and if it goes well we can adapt it to other reducers